functional/unit_action_test.go

```diff
@@ -73,48 +76,47 @@ func TestUnitSubmit(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	unitFile := "fixtures/units/hello.service"
+
 	// submit a unit and assert it shows up
-	if _, _, err := cluster.Fleetctl(m, "submit", "fixtures/units/hello.service"); err != nil {
+	_, _, err = cluster.Fleetctl(m, "submit", unitFile)
+	if err != nil {
 		t.Fatalf("Unable to submit fleet unit: %v", err)
 	}
-	stdout, _, err := cluster.Fleetctl(m, "list-units", "--no-legend")
```

Comment:
@antrik 
> no, it was indeed completely wrong: list-units returns an empty string; but that's still counts as one line, so the test "passed"...

Yes, that was exactly what I wanted to say. That's why I said yesterday that this part has been wrong since the beginning. I think the underlying reason was that ``len(strings.Split("", "\n"))`` returns 1, not 0 as expected.

So my solution to get it working was adding additional conditions like "``expectedUnits == 0 && len(outListUnitFiles) == 0``" in each check function. Still somehow nasty, but it's working for now. Afterwards, we should do some code auditing in the entire tree to find such errors. But anyway that would be over the scope of this PR.